Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1225,6 +1225,12 @@ static String mutationObserverNotificationString()
     return std::nullopt;
 }
 
+std::optional<PDFDocumentLayout::PageIndex> UnifiedPDFPlugin::indexForCurrentPageInView() const
+{
+    auto centerInDocumentSpace = convertFromPluginToDocument(flooredIntPoint(size() / 2));
+    return pageIndexForDocumentPoint(centerInDocumentSpace);
```

Comment:
Does this find the nearest page, or does it just fail sometimes?

Reply:
Yeah so I couldn't actually get something to reproduce locally but after taking a closer look at the code I'm pretty sure it can fail. I think the document layout bounds do not include margins which is the problem. 

Also it turns out this does not work well for two page display modes so I'm going to fix both in a follow up patch. Made a note in the commit message stating two page display has issues